### ISSUE TITLE
Loss diverges when training large hourglass architectures

### ISSUE THREAD

**Issue by PanosAntoniadis** [lucidrains/hourglass-transformer-pytorch#5]
Thanks for providing the implementation for the model in pytorch! 

I am struggling to train a hourglass model with more parameters than the ones provided in the paper.

When I using a 144M model with 
- Layers: 4 @ 1, 4 @ 8, 4 @ 8, 4 @ 8, 4 @ 1
- token_dim: 768
- 8 heads - 64 dimension each

the train loss curve looks normal.

<img width="1264" height="664" alt="Image" src="https://github.com/user-attachments/assets/e523281b-709c-4dee-ad91-c692eeadce3a" />

However, with exactly the same setup (data, trainer etc) when I try to increase the number of layers and the token dimension to end up with a 450M model:
- Layers: 7 @ 1, 6 @ 8, 5 @ 8, 6 @ 8, 7 @ 1
- token_dim: 1024
- 16 heads - 64 dimension each 

the train loss curve completely diverges after 40k steps. 

<img width="1264" height="664" alt="Image" src="https://github.com/user-attachments/assets/2fc6f0cc-fa6a-4235-a55e-8fd479a8cc4f" />

In both experiment I am using gradient clip to 1 and a cosine lr scheduler with warmup to avoid these spikes in the loss. Can you help me on what I can try in order to increase the number of parameters and avoid the issue above?
